src/Navbar.js

```diff
@@ -57,7 +58,36 @@ const propTypes = {
    * @controllable navExpanded
    */
   onToggle: React.PropTypes.func,
-
+  /**
+   * A callback fired when a `<NavItem>` grandchild is selected inside a child
+   * `<Nav>`. Should be used to execute complex toggling or other miscellaneous
+   * actions desired after selecting a `<NavItem>`. Does nothing if no `<Nav>`
+   * & `<NavItem>`/`<MenuItem>` children exist. The callback is called with an
+   * eventKey, which is a prop from the `<NavItem>`, and an event.
+   *
+   * ```js
+   * function (
+   * 	Any eventKey,
+   * 	SyntheticEvent event?
+   * )
+   * ```
+   *
+   * For basic closing behavior after all `<NavItem>` onSelect events in mobile
+   * viewports, try using closeOnSelect.
+   * Note: If you are manually closing the navbar using this `OnSelect` prop,
+   * ensure that you are setting `expanded` to false and not *toggling* between
+   * true and false.
+   */
+  onSelect: React.PropTypes.func,
+  /**
+   * Sets `expanded` to false after the onSelect event of a child of `<Nav>`
+   * (such as `<NavItem>` or `<MenuItem>`). Does nothing if no `<Nav>` or
+   * `<NavItem>`/`<MenuItem>` children exist.
```

Comment:
Working off of what I brainstormed in the comment above, how's this?
"
Sets `expanded` to `false` after the onSelect event of a descendant of a child `<Nav>`. Does nothing if no `<Nav>` or `<Nav>` descendants exist.
"